### PR TITLE
use block.timestamp for dates

### PR DESCRIPTION
**Pull request by kangarang**
this pr includes an experimental commit which considers the latest `block.timestamp` as "now". this is useful when discovering locked/unlocked/projected balances, and the additional changes made to the `time-travel.js` script in #117 can be used in conjunction with the changes in this pr to see differences in locked/unlocked/projected token balances.

### CODE REVIEW

**Comment by kangarang** [comment]
related: #113

File: client/components/SlateSidebar.tsx [hunk 1]
```diff
@@ -55,7 +55,7 @@ const SlateSidebar = ({ slate, requiredStake, currentBallot }: IStakeSidebarProp
 
   // Calculate the extended deadline from now and the start of the commit period,
   // assuming you were to stake right now
-  const now = timestamp();
+  const now = timestamp(currentBallot.bts);
```
Comment (akuanti) [inline client/components/SlateSidebar.tsx]: You don't need to call `timestamp()` here, since this is already a timestamp.

File: client/utils/datetime.ts [hunk 2]
```diff
@@ -15,8 +15,9 @@ export const tsToDeadline = (unixTimestamp: number, customTimezone?: string): st
   return deadline.tz(customTimezone).format(fmt);
 };
 
-export const dateHasPassed = (uts: number) => isAfter(new Date(), new Date(uts * 1000));
+export const dateHasPassed = (uts: number, blockTimestamp: number) =>
+  isAfter(new Date(blockTimestamp * 1000), new Date(uts * 1000));
 
-export const timestamp = (): number => {
-  return Math.floor(new Date().getTime() / 1000);
+export const timestamp = (blockTimestamp: number): number => {
```
Comment (akuanti) [inline client/utils/datetime.ts]: This gives you back the number you pass in, so we can delete it or put it back the way it was.

File: packages/panvala-utils/voting/index.test.js [hunk 3]
```diff
@@ -3,11 +3,11 @@ const { voting } = require('../index.js');
 
 test('create a commit hash', () => {
   const votes = {
-    0: {
+    '0xd115bffabbdd893a6f7cea402e7338643ced44a6': {
```
Comment (akuanti) [inline packages/panvala-utils/voting/index.test.js]: Revert these changes so we don't overwrite the newer changes in `develop`.

**Comment by akuanti** [comment]
Don't merge until #113 is merged. Also, consider that we don't have access to `block.timestamp` if the user doesn't have MetaMask installed and connected.

**Comment by kangarang** [comment]
i think we can close this without merging. it's not too useful. if we feel the need to re-open it, let's do so at that time.
